deploy/helm/sumologic/values.yaml

```diff
@@ -4370,37 +4370,42 @@ otellogs:
             timestamp:
               parse_from: time
               layout: '%Y-%m-%dT%H:%M:%S.%LZ'
 
-          ## This operator is currently disabled as it adds additional newlines
-          ## https://github.com/open-telemetry/opentelemetry-log-collection/issues/314
-          #
           ## merge-split-lines stitches back together log lines split by Docker logging driver.
           # Input Body (JSON): { "log": "2001-02-03 04:05:06 very long li", "stream": "stdout" }
           # Input Body (JSON): { "log": "ne that was split by the logging driver\n", "stream": "stdout" }
           # Output Body (JSON): { "log": "2001-02-03 04:05:06 very long line that was split by the logging driver\n", "stream": "stdout" }
-          # - id: merge-split-lines
-          #   type: recombine
-          #   combine_field: log
-          #   is_last_entry: $$body.log matches "\n$"
-
-          ## This operator is currently disabled due to the following issues:
-          ## - Additional newlines being added between the merged logs:
-          ## https://github.com/open-telemetry/opentelemetry-log-collection/issues/314
-          ## - No flushing: https://github.com/open-telemetry/opentelemetry-log-collection/issues/306
-          #
+          - id: merge-split-lines
+            type: recombine
+            output: merge-multiline-logs
+            combine_field: log
+            is_last_entry: $$body.log matches "\n$"
+          
+          ## merge-cri-lines stitches back together log lines split by CRI logging drivers.
+          # Input Body (JSON): { "log": "2001-02-03 04:05:06 very long li", "logtag": "P" }
+          # Input Body (JSON): { "log": "ne that was split by the logging driver", "logtag": "F" }
+          # Output Body (JSON): { "log": "2001-02-03 04:05:06 very long line that was split by the logging driver\n", "stream": "stdout" }
+          - id: merge-cri-logs
+            type: recombine
+            output: merge-multiline-logs
+            combine_field: log
+            combine_with: ""
+            is_last_entry: "$body.logtag == 'F'"
+            overwrite_with: newest
+
           ## merge-multiline-logs merges incoming log records into multiline logs.
           # Input Body (JSON): { "log": "2001-02-03 04:05:06 first line\n", "stream": "stdout" }
           # Input Body (JSON): { "log": "  second line\n", "stream": "stdout" }
           # Input Body (JSON): { "log": "  third line\n", "stream": "stdout" }
           # Output Body (JSON): { "log": "2001-02-03 04:05:06 first line\n  second line\n  third line\n", "stream": "stdout" }
-          # - id: merge-multiline-logs
-          #   type: recombine
-          #   combine_field: log
-          #   is_first_entry: $$body.log matches "^[^\\s]"
+          - id: merge-multiline-logs
+            type: recombine
+            output: extract-metadata-from-filepath
+            combine_field: log
+            is_first_entry: $$body.log matches "^[^\\s]"
```

Comment:
@swiatekm-sumo I just uncommented this, but my question is, why is this different from this what we do in fluent-bit?
(in fluent bit we matches by date format)

Reply:
@astencel-sumo you wrote this originally, I just copied it mindlessly.